source/simulator/introspection.cc

```diff
@@ -332,6 +332,39 @@ namespace aspect
 
 
 
+  template <int dim>
+  std::vector<unsigned int>
+  Introspection<dim>::get_compositional_field_base_element_indices() const
+  {
+    // We are assigning base elements in order, so the first compositional field
+    // gives us the first base element index. Then we find the largest index
+    // in the vector. This is necessary, because the fields could have type A,B,A.
+    std::vector<unsigned int> result;
+    const unsigned int first = this->base_elements.compositional_fields[0];
+    const unsigned int last = *std::max_element(this->base_elements.compositional_fields.begin(), this->base_elements.compositional_fields.end());
+    for (unsigned int i = first; i<=last; ++i)
+      result.emplace_back(i);
+    return result;
+  }
+
+
+
+  template <int dim>
+  std::vector<unsigned int>
+  Introspection<dim>::get_compositional_field_indices_with_base_element(const unsigned int base_element_index) const
+  {
+    std::vector<unsigned int> result;
+
```

Comment:
This is where you want to check that `base_element_index` is ever used. You can probably do that by making sure that the output vector is non-empty.